cli/src/memo.rs

```diff
@@ -0,0 +1,22 @@
+use solana_sdk::instruction::Instruction;
+use solana_sdk::pubkey::Pubkey;
+use std::str::FromStr;
+
+pub trait WithMemo {
+    fn with_memo(self, memo: Option<&String>) -> Self;
+}
+
+impl WithMemo for Vec<Instruction> {
+    fn with_memo(mut self, memo: Option<&String>) -> Self {
+        if let Some(memo) = &memo {
+            let memo_ix = Instruction {
```

Comment:
Could use `spl_memo::build_memo()` here to avoid replicode.  Also `spl_memo::id()` throughout

Reply:
+1